cdap-watchdog/src/main/java/co/cask/cdap/logging/framework/CDAPLogAppender.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.framework;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.AppenderBase;
+import co.cask.cdap.logging.write.FileMetaDataManager;
+import com.google.common.base.Throwables;
+import com.google.common.io.Closeables;
+import org.apache.avro.Schema;
+import org.apache.twill.filesystem.LocationFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Flushable;
+import java.io.IOException;
+
+/**
+ * Log Appender implementation for CDAP Log framework
+ */
+public class CDAPLogAppender extends AppenderBase<ILoggingEvent> implements Flushable {
+  private static final Logger LOG = LoggerFactory.getLogger(CDAPLogAppender.class);
+
+  private boolean isStarted;
+  private final AvroFileManager avroFileManager;
+
+  /**
+   * TODO: start a separate cleanup thread to remove files that has passed the TTL
+   */
+
+  /**
+   * Constructs an AvroFileWriter object.
+   * @param fileMetaDataManager used to store file meta data.
+   * @param locationFactory the location factory
+   * @param schema schema of the Avro data to be written.
+   * @param syncIntervalBytes the approximate number of uncompressed bytes to write in each block.
+   * @param maxFileLifetimeMs files that are older than maxFileLifetimeMs will be closed.
+   */
+  public CDAPLogAppender(FileMetaDataManager fileMetaDataManager, LocationFactory locationFactory,
```

Comment:
I don't think you can instantiate an `Appender` like this. For configurations that come from `logback.xml`, you need to provide public setting methods. For objects that come from CDAP (e.g. `LocationFactory`), you have to use Guice method injection or field injection.